website/addons/github/model.py

```diff
@@ -76,10 +76,7 @@ def revoke_remote_oauth_access(self, external_account):
     # Assumes oldest connected account is primary.
     @property
     def public_id(self):
-        gh_accounts = [
-            a for a in self.owner.external_accounts.all()
-            if a.provider == self.oauth_provider.short_name
-        ]
+        gh_accounts = self.owner.external_accounts.filter(provider=self.oauth_provider.short_name)
         if gh_accounts:
             return gh_accounts[0].display_name
```

Comment:
Yeah, the django queryset's are indexable and sliceable. They will be evaluated when sliced though so you'll have a python object, not a queryset anymore.

You can't do:
```python
Node.objects.all()[0:20].order_by('title')
```
But you can do:
```python
Node.objects.all().order_by('title')[0:20]
```